Sources/Widgets/Core/StateBuilder/index.js

```diff
@@ -25,6 +26,7 @@ const MIXINS = {
   direction,
   manipulator,
   name,
+  normal,
```

Comment:
isn't it direction/orientation in charge of being the "normal" ? @agirault @floryst

Reply:
The problem I had with `direction` was that it was always positive, regardless of the normal's orientation.

Reply:
I think it is a mistake then. To me, a `direction` is:  `orientation` + `sign`. 

What do you think @floryst  ?

Reply:
It seems here that `direction` is currently expected to be two 0s and a 1. See line 68:

`const slicingMode = state.getDirection().indexOf(1);`

That does seem strange.

Reply:
`direction` is typically interpreted as `normal`, especially when defining planes. `RectangleContextRepresentation` shouldn't be assuming that the direction is positive and axis-aligned. However, I see that the representation is assuming IJK oriented planes, so that'll explain the assumption that the direction is axis-aligned.

Reply:
The widget can use `direction` however they so please, so some widgets might assume constraints on `direction`. Though it would be nice to have widgets that aren't constrained to be axis-aligned when they don't need to be.

Reply:
Thanks @floryst 

So I guess the easiest to do for now is keep `direction` instead of `normal`.  Just convert `state.getDirection().indexOf(1)` to `state.getDirection().indexOf(state.getDirection().reduce((max, comp) Math.max(max, Math.abs(comp)));`  or something like that...

Reply:
Some of it was already mentioned, but:
- `direction` should be the same as `normal`
- it isn't right now because it's limited to some slicing modes
- `orientation` is not just the `direction/normal`, but a full rotation (so also `up` and `right` vectors). We can use `orientation` instead of `direction` to properly define arbitrary orientations. That's required to keep the edges of the shapes parallel with the edges of the camera view when the camera plane isn't XY, YZ, or XZ.

#1422 should handle all of that fine for the rectangle widget.